fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -1563,59 +1563,120 @@ public void replayAddPartition(PartitionPersistInfo info) throws DdlException {
-                                        " The partition [" + partitionName +
-                                        "] cannot be dropped. If you want to forcibly drop(cannot be recovered)," +
-                                        " please use \"DROP PARTITION <partition> FORCE\".");
+        if (clause.hasMultiPartitions()) {
+            PartitionDesc partitionDesc = clause.getPartitionDesc();
+            if (partitionDesc instanceof MultiRangePartitionDesc) {
+
+                if (!(partitionInfo instanceof RangePartitionInfo)) {
+                    throw new DdlException("Batch deletion of partitions only support range partition tables.");
+                }
+                RangePartitionInfo rangePartitionInfo = (RangePartitionInfo) partitionInfo;
+
+                List<Column> partitionColumns = rangePartitionInfo.getPartitionColumns();
+                if (partitionColumns.size() != 1) {
+                    throw new DdlException("Alter batch drop partition only support single range column.");
+                }
+
+                Column firstPartitionColumn = partitionColumns.get(0);
+                MultiRangePartitionDesc multiRangePartitionDesc = (MultiRangePartitionDesc) partitionDesc;
+                boolean isAutoPartitionTable = false;
+                if (partitionInfo.getType() == PartitionType.EXPR_RANGE) {
+                    isAutoPartitionTable = true;
+                    ExpressionRangePartitionInfo exprRangePartitionInfo = (ExpressionRangePartitionInfo) partitionInfo;
+                    Expr expr = exprRangePartitionInfo.getPartitionExprs().get(0);
+                    if (expr instanceof FunctionCallExpr) {
+                        FunctionCallExpr functionCallExpr = (FunctionCallExpr) expr;
+                        checkAutoPartitionTableLimit(functionCallExpr, multiRangePartitionDesc);
                     }
                 }
+                Map<String, String> properties = clause.getProperties();
+                if (properties == null) {
+                    properties = Maps.newHashMap();
+                }
+                if (tableProperties != null && tableProperties.containsKey(DynamicPartitionProperty.START_DAY_OF_WEEK)) {
+                    properties.put(DynamicPartitionProperty.START_DAY_OF_WEEK,
+                            tableProperties.get(DynamicPartitionProperty.START_DAY_OF_WEEK));
+                }
+                PartitionConvertContext context = new PartitionConvertContext();
+                context.setAutoPartitionTable(isAutoPartitionTable);
+                context.setFirstPartitionColumnType(firstPartitionColumn.getType());
+                context.setProperties(properties);
+                context.setTempPartition(clause.isTempPartition());
+                List<SingleRangePartitionDesc> singleRangePartitionDescs = multiRangePartitionDesc.convertToSingle(context);
+                partitionNames.addAll(singleRangePartitionDescs.stream()
```

Comment:
Shell we take consideration of the number of partitions? If it's large, will it be OK?

Reply:
The time complexity of this method is O(N), and I believe there are no performance issues within a reasonable time frame.